test failure: Eclipse SmartHome Automation Core Module Tests

Perhaps caused by some timing / async handling stuff.
Needs further inspection...

* <code>git rev-parse HEAD</code>: 34528535baac65a74100703fbe0a50a4314729d9
* <code>git status</code>
```text
On branch master
Your branch is up-to-date with 'origin/master'.
Untracked files:
  (use "git add <file>..." to include in what will be committed)

	distribution/smarthome/userdata/

nothing added to commit but untracked files present (use "git add" to track)
```

```text
[INFO] ------------------------------------------------------------------------
[INFO] Building Eclipse SmartHome Automation Core Module Tests 0.8.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ org.eclipse.smarthome.automation.module.core.test ---
[INFO] Deleting /home/maggu2810/bin/pkgs/eclipse/smarthome-master/git/smarthome/bundles/automation/org.eclipse.smarthome.automation.module.core.test/target
[INFO] 
[INFO] --- tycho-packaging-plugin:0.23.1:build-qualifier (default-build-qualifier) @ org.eclipse.smarthome.automation.module.core.test ---
[INFO] The project's OSGi version is 0.8.0.201601290907
[INFO] 
[INFO] --- tycho-packaging-plugin:0.23.1:validate-id (default-validate-id) @ org.eclipse.smarthome.automation.module.core.test ---
[INFO] 
[INFO] --- tycho-packaging-plugin:0.23.1:validate-version (default-validate-version) @ org.eclipse.smarthome.automation.module.core.test ---
[INFO] 
[INFO] --- xtend-maven-plugin:2.6.2:compile (default) @ org.eclipse.smarthome.automation.module.core.test ---
[INFO] No sources to compile in '/home/maggu2810/bin/pkgs/eclipse/smarthome-master/git/smarthome/bundles/automation/org.eclipse.smarthome.automation.module.core.test/src/test/groovy'
[INFO] 
[INFO] --- build-helper-maven-plugin:1.8:add-source (add-source) @ org.eclipse.smarthome.automation.module.core.test ---
[INFO] Source directory: /home/maggu2810/bin/pkgs/eclipse/smarthome-master/git/smarthome/bundles/automation/org.eclipse.smarthome.automation.module.core.test/src/main/groovy added.
[INFO] 
[INFO] --- maven-resources-plugin:2.4.3:resources (default-resources) @ org.eclipse.smarthome.automation.module.core.test ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/maggu2810/bin/pkgs/eclipse/smarthome-master/git/smarthome/bundles/automation/org.eclipse.smarthome.automation.module.core.test/src/main/resources
[INFO] 
[INFO] --- tycho-compiler-plugin:0.23.1:compile (default-compile) @ org.eclipse.smarthome.automation.module.core.test ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:compile (default) @ org.eclipse.smarthome.automation.module.core.test ---
[INFO] Changes detected - recompiling the module!
[INFO] Using Groovy-Eclipse compiler to compile both Java and Groovy files
[INFO] 
[INFO] --- xtend-maven-plugin:2.6.2:xtend-install-debug-info (default) @ org.eclipse.smarthome.automation.module.core.test ---
[WARNING] Directory /home/maggu2810/bin/pkgs/eclipse/smarthome-master/git/smarthome/bundles/automation/org.eclipse.smarthome.automation.module.core.test/src/main/java is empty. Can't process.
[WARNING] Directory /home/maggu2810/bin/pkgs/eclipse/smarthome-master/git/smarthome/bundles/automation/org.eclipse.smarthome.automation.module.core.test/xtend-gen is empty. Can't process.
[WARNING] Directory /home/maggu2810/bin/pkgs/eclipse/smarthome-master/git/smarthome/bundles/automation/org.eclipse.smarthome.automation.module.core.test/src/main/groovy is empty. Can't process.
[INFO] Installing Xtend files into 0 class files as secondary (via SMAP) debug sources in: /home/maggu2810/bin/pkgs/eclipse/smarthome-master/git/smarthome/bundles/automation/org.eclipse.smarthome.automation.module.core.test/target/classes
[INFO] 
[INFO] --- xtend-maven-plugin:2.6.2:testCompile (default) @ org.eclipse.smarthome.automation.module.core.test ---
[INFO] skip compiling sources because the configured directory '[/home/maggu2810/bin/pkgs/eclipse/smarthome-master/git/smarthome/bundles/automation/org.eclipse.smarthome.automation.module.core.test/src/test/java]' does not exists.
[INFO] 
[INFO] --- build-helper-maven-plugin:1.8:add-test-source (add-test-source) @ org.eclipse.smarthome.automation.module.core.test ---
[INFO] Test Source directory: /home/maggu2810/bin/pkgs/eclipse/smarthome-master/git/smarthome/bundles/automation/org.eclipse.smarthome.automation.module.core.test/src/test/groovy added.
[INFO] 
[INFO] --- maven-resources-plugin:2.4.3:testResources (default-testResources) @ org.eclipse.smarthome.automation.module.core.test ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/maggu2810/bin/pkgs/eclipse/smarthome-master/git/smarthome/bundles/automation/org.eclipse.smarthome.automation.module.core.test/src/test/resources
[INFO] 
[INFO] --- xtend-maven-plugin:2.6.2:xtend-test-install-debug-info (default) @ org.eclipse.smarthome.automation.module.core.test ---
[WARNING] Directory /home/maggu2810/bin/pkgs/eclipse/smarthome-master/git/smarthome/bundles/automation/org.eclipse.smarthome.automation.module.core.test/src/test/java is empty. Can't process.
[WARNING] Directory /home/maggu2810/bin/pkgs/eclipse/smarthome-master/git/smarthome/bundles/automation/org.eclipse.smarthome.automation.module.core.test/xtend-gen is empty. Can't process.
[INFO] Installing Xtend files into 0 class files as secondary (via SMAP) debug sources in: /home/maggu2810/bin/pkgs/eclipse/smarthome-master/git/smarthome/bundles/automation/org.eclipse.smarthome.automation.module.core.test/target/test-classes
[INFO] 
[INFO] --- tycho-source-plugin:0.23.1:plugin-source (plugin-source) @ org.eclipse.smarthome.automation.module.core.test ---
[WARNING] bundle localization file /home/maggu2810/bin/pkgs/eclipse/smarthome-master/git/smarthome/bundles/automation/org.eclipse.smarthome.automation.module.core.test/plugin.properties not found
[INFO] Building jar: /home/maggu2810/bin/pkgs/eclipse/smarthome-master/git/smarthome/bundles/automation/org.eclipse.smarthome.automation.module.core.test/target/org.eclipse.smarthome.automation.module.core.test-0.8.0-SNAPSHOT-sources.jar
[INFO] 
[INFO] --- target-platform-configuration:0.23.1:target-platform (default-target-platform) @ org.eclipse.smarthome.automation.module.core.test ---
[INFO] 
[INFO] --- tycho-packaging-plugin:0.23.1:package-plugin (default-package-plugin) @ org.eclipse.smarthome.automation.module.core.test ---
[INFO] Building jar: /home/maggu2810/bin/pkgs/eclipse/smarthome-master/git/smarthome/bundles/automation/org.eclipse.smarthome.automation.module.core.test/target/org.eclipse.smarthome.automation.module.core.test-0.8.0-SNAPSHOT.jar
[INFO] 
[INFO] --- tycho-p2-plugin:0.23.1:p2-metadata-default (default-p2-metadata-default) @ org.eclipse.smarthome.automation.module.core.test ---
[INFO] 
[INFO] --- tycho-surefire-plugin:0.23.1:test (default-test) @ org.eclipse.smarthome.automation.module.core.test ---
[INFO] Expected eclipse log file: /home/maggu2810/bin/pkgs/eclipse/smarthome-master/git/smarthome/bundles/automation/org.eclipse.smarthome.automation.module.core.test/target/work/data/.metadata/.log
[INFO] Command line:
	[/opt/oracle-jdk-bin-1.8.0.66/jre/bin/java, -Dosgi.noShutdown=false, -Dosgi.os=linux, -Dosgi.ws=gtk, -Dosgi.arch=x86_64, -Dosgi.clean=true, -jar, /home/maggu2810/.m2/repository/p2/osgi/bundle/org.eclipse.equinox.launcher/1.3.0.v20140415-2008/org.eclipse.equinox.launcher-1.3.0.v20140415-2008.jar, -data, /home/maggu2810/bin/pkgs/eclipse/smarthome-master/git/smarthome/bundles/automation/org.eclipse.smarthome.automation.module.core.test/target/work/data, -install, /home/maggu2810/bin/pkgs/eclipse/smarthome-master/git/smarthome/bundles/automation/org.eclipse.smarthome.automation.module.core.test/target/work, -configuration, /home/maggu2810/bin/pkgs/eclipse/smarthome-master/git/smarthome/bundles/automation/org.eclipse.smarthome.automation.module.core.test/target/work/configuration, -application, org.eclipse.tycho.surefire.osgibooter.headlesstest, -testproperties, /home/maggu2810/bin/pkgs/eclipse/smarthome-master/git/smarthome/bundles/automation/org.eclipse.smarthome.automation.module.core.test/target/surefire.properties]
10:08:47.784 [Start Level: Equinox Container: e0378be7-67c6-0015-1437-f6e393ce414d] DEBUG o.e.s.core.internal.CoreActivator - Core bundle has been started.
10:08:47.932 [Start Level: Equinox Container: e0378be7-67c6-0015-1437-f6e393ce414d] DEBUG o.e.s.c.a.i.AutoUpdateActivator - AutoUpdate binding has been started.
10:08:48.266 [Component Resolve Thread] DEBUG o.e.s.core.common.ThreadPoolManager - Created thread pool 'ConfigStatusService' with size 5-10
10:08:48.269 [Start Level: Equinox Container: e0378be7-67c6-0015-1437-f6e393ce414d] DEBUG o.e.s.a.m.core.internal.Activator - started bundle automation.module

-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running org.eclipse.smarthome.automation.module.core.internal.RuntimeRuleTest
10:08:48.789 [main] DEBUG o.e.s.c.c.registry.AbstractRegistry - Provider 'com.sun.proxy.$Proxy8' has been added.
10:08:48.799 [main] DEBUG o.e.s.a.i.c.p.AutomationResourceBundlesEventQueue - Process bundle event 1, for automation bundle 'org.eclipse.smarthome.automation.module.core' 
10:08:48.800 [main] DEBUG o.e.s.a.i.c.p.AutomationResourceBundlesEventQueue - Process bundle event 1, for automation bundle 'org.eclipse.smarthome.automation.module.core.test' 
10:08:48.804 [main] DEBUG o.e.s.c.c.registry.AbstractRegistry - Provider 'org.eclipse.smarthome.automation.core.internal.ManagedRuleProvider' has been added.
10:08:48.806 [main] DEBUG o.e.s.c.c.registry.AbstractRegistry - Provider 'org.eclipse.smarthome.core.items.ManagedItemProvider' has been added.
10:08:48.827 [Automation Provider Processing Queue] ERROR o.e.s.a.i.c.p.ModuleTypeResourceBundleProvider - ModuleType parsing error!
org.eclipse.smarthome.automation.parser.ParsingException: [Module Type] com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_ARRAY but was BEGIN_OBJECT at line 28 column 22 path $.conditions[0].configDescriptions[2].options

	at org.eclipse.smarthome.automation.parser.gson.internal.ModuleTypeGSONParser.parse(ModuleTypeGSONParser.java:53) ~[na:na]
	at org.eclipse.smarthome.automation.internal.core.provider.ModuleTypeResourceBundleProvider.importData(ModuleTypeResourceBundleProvider.java:226) ~[org.eclipse.smarthome.automation.providers-0.8.0-SNAPSHOT.jar:na]
	at org.eclipse.smarthome.automation.internal.core.provider.AbstractResourceBundleProvider.processAutomationProvider(AbstractResourceBundleProvider.java:315) [org.eclipse.smarthome.automation.providers-0.8.0-SNAPSHOT.jar:na]
	at org.eclipse.smarthome.automation.internal.core.provider.AutomationResourceBundlesEventQueue.processBundleChanged(AutomationResourceBundlesEventQueue.java:330) [org.eclipse.smarthome.automation.providers-0.8.0-SNAPSHOT.jar:na]
	at org.eclipse.smarthome.automation.internal.core.provider.AutomationResourceBundlesEventQueue.run(AutomationResourceBundlesEventQueue.java:154) [org.eclipse.smarthome.automation.providers-0.8.0-SNAPSHOT.jar:na]
	at java.lang.Thread.run(Thread.java:745) [na:1.8.0_66]
10:08:48.831 [Automation Provider Processing Queue] DEBUG o.e.s.a.i.c.p.RuleResourceBundleImporter - Parse rules from bundle 'org.eclipse.smarthome.automation.module.core' 
10:08:48.831 [Automation Provider Processing Queue] DEBUG o.e.s.a.i.c.p.RuleResourceBundleImporter - Parse rules from bundle 'org.eclipse.smarthome.automation.module.core.test' 
10:08:48.839 [main] INFO  o.e.s.a.m.c.h.CompareConditionHandler - unsupported type for compare condition: null (null)
10:08:48.840 [main] INFO  o.e.s.a.m.c.h.CompareConditionHandler - unsupported type for compare condition: class org.eclipse.smarthome.core.library.types.OnOffType
10:08:48.846 [main] INFO  o.e.s.a.m.c.h.CompareConditionHandler - unsupported type for compare condition: null (null)
10:08:48.846 [main] INFO  o.e.s.a.m.c.h.CompareConditionHandler - unsupported type for compare condition: class org.eclipse.smarthome.core.items.events.ItemStateEvent
10:08:48.861 [main] DEBUG o.e.s.c.c.registry.AbstractRegistry - Provider '$Proxy8' has been removed.
10:08:48.862 [main] DEBUG o.e.s.c.c.registry.AbstractRegistry - Provider 'ManagedRuleProvider' has been removed.
10:08:48.864 [main] DEBUG o.e.s.c.c.registry.AbstractRegistry - Provider 'ManagedItemProvider' has been removed.
10:08:48.874 [main] DEBUG o.e.s.c.c.registry.AbstractRegistry - Provider 'com.sun.proxy.$Proxy8' has been added.
10:08:48.876 [main] DEBUG o.e.s.c.c.registry.AbstractRegistry - Provider 'org.eclipse.smarthome.automation.core.internal.ManagedRuleProvider' has been added.
10:08:48.876 [main] DEBUG o.e.s.c.c.registry.AbstractRegistry - Provider 'org.eclipse.smarthome.core.items.ManagedItemProvider' has been added.
10:08:51.930 [main] DEBUG o.e.s.c.c.registry.AbstractRegistry - Provider '$Proxy8' has been removed.
10:08:51.930 [main] DEBUG o.e.s.c.c.registry.AbstractRegistry - Provider 'ManagedRuleProvider' has been removed.
10:08:51.930 [main] DEBUG o.e.s.c.c.registry.AbstractRegistry - Provider 'ManagedItemProvider' has been removed.
10:08:51.937 [main] DEBUG o.e.s.c.c.registry.AbstractRegistry - Provider 'com.sun.proxy.$Proxy8' has been added.
10:08:51.938 [main] DEBUG o.e.s.c.c.registry.AbstractRegistry - Provider 'org.eclipse.smarthome.automation.core.internal.ManagedRuleProvider' has been added.
10:08:51.939 [main] DEBUG o.e.s.c.c.registry.AbstractRegistry - Provider 'org.eclipse.smarthome.core.items.ManagedItemProvider' has been added.
10:08:51.960 [main] INFO  o.e.s.a.m.c.internal.RuntimeRuleTest - Rule created: myRule21-2127980933_COMPOSITE
10:08:51.971 [main] DEBUG o.e.s.a.core.internal.RuleEngine - Added rule 'myRule21-2127980933_COMPOSITE'
10:08:51.974 [main] DEBUG o.e.s.a.c.i.c.CompositeModuleHandlerFactory - create composite module:org.eclipse.smarthome.automation.core.internal.RuntimeCondition@1bb1fde8, of rule: myRule21-2127980933_COMPOSITE
10:08:51.974 [main] DEBUG o.e.s.a.c.i.c.CompositeModuleHandlerFactory - create composite module:org.eclipse.smarthome.automation.Condition@15eebbff, of rule: myRule21-2127980933_COMPOSITE
10:08:51.975 [main] DEBUG o.e.s.a.c.i.c.CompositeModuleHandlerFactory - Set handler to composite module:org.eclipse.smarthome.automation.Condition@15eebbff -> org.eclipse.smarthome.automation.core.internal.composite.CompositeConditionHandler@3e34ace1
10:08:51.975 [main] DEBUG o.e.s.a.c.i.c.CompositeModuleHandlerFactory - Set handler to composite module:org.eclipse.smarthome.automation.core.internal.RuntimeCondition@1bb1fde8 -> org.eclipse.smarthome.automation.core.internal.composite.CompositeConditionHandler@62fe6067
10:08:51.976 [main] DEBUG o.e.s.a.c.i.c.CompositeModuleHandlerFactory - create composite module:org.eclipse.smarthome.automation.core.internal.RuntimeTrigger@305f7627, of rule: myRule21-2127980933_COMPOSITE
10:08:51.978 [main] DEBUG o.e.s.a.c.i.c.CompositeModuleHandlerFactory - Set handler to composite module:org.eclipse.smarthome.automation.core.internal.RuntimeTrigger@305f7627 -> org.eclipse.smarthome.automation.core.internal.composite.CompositeTriggerHandler@7bc10d84
10:08:51.979 [main] DEBUG o.e.s.c.c.r.AbstractManagedProvider - Added new element myRule21-2127980933_COMPOSITE to ManagedRuleProvider.
10:08:52.011 [EventAdmin Async Event Dispatcher Thread] DEBUG o.e.s.core.common.ThreadPoolManager - Created thread pool 'safeCall' with size 5-10
10:08:52.016 [pool-2-thread-1] DEBUG o.e.s.a.m.c.h.ItemStateConditionHandler - ItemStateCondition 'ItemStateCondition5'checking if myPresenceItem3 (State=ON) = ON
10:08:52.016 [pool-2-thread-1] DEBUG o.e.s.a.m.c.h.ItemStateConditionHandler - ConditionSatisfied --> true
10:08:52.017 [ESH-safeCall-5] INFO  o.e.s.a.m.c.internal.RuntimeRuleTest - RuleEvent: smarthome/rules/myRule21-2127980933_COMPOSITE/state
10:08:52.017 [pool-2-thread-1] DEBUG o.e.s.a.m.c.h.ItemPostCommandActionHandler - Executing ItemPostCommandAction on Item myLampItem3 with Command ON
10:08:52.017 [pool-2-thread-1] DEBUG o.e.s.a.core.internal.RuleEngine - The rule 'myRule21-2127980933_COMPOSITE' is executed.
10:08:52.018 [ESH-safeCall-5] INFO  o.e.s.a.m.c.internal.RuntimeRuleTest - RuleEvent: smarthome/rules/myRule21-2127980933_COMPOSITE/state
10:08:52.067 [main] DEBUG o.e.s.c.c.registry.AbstractRegistry - Provider '$Proxy8' has been removed.
10:08:52.069 [main] DEBUG o.e.s.c.c.registry.AbstractRegistry - Provider 'ManagedRuleProvider' has been removed.
10:08:52.069 [main] DEBUG o.e.s.c.c.registry.AbstractRegistry - Provider 'ManagedItemProvider' has been removed.
10:08:52.072 [main] DEBUG o.e.s.c.c.registry.AbstractRegistry - Provider 'com.sun.proxy.$Proxy8' has been added.
10:08:52.074 [main] DEBUG o.e.s.c.c.registry.AbstractRegistry - Provider 'org.eclipse.smarthome.automation.core.internal.ManagedRuleProvider' has been added.
10:08:52.075 [main] DEBUG o.e.s.c.c.registry.AbstractRegistry - Provider 'org.eclipse.smarthome.core.items.ManagedItemProvider' has been added.
10:08:52.075 [main] INFO  o.e.s.a.m.c.internal.RuntimeRuleTest - Rule created: myRule21-1546347186
10:08:52.076 [main] DEBUG o.e.s.a.core.internal.RuleEngine - Added rule 'myRule21-1546347186'
10:08:52.076 [main] DEBUG o.e.s.a.c.i.c.CompositeModuleHandlerFactory - create composite module:org.eclipse.smarthome.automation.core.internal.RuntimeCondition@6ca0256d, of rule: myRule21-1546347186
10:08:52.076 [main] DEBUG o.e.s.a.c.i.c.CompositeModuleHandlerFactory - create composite module:org.eclipse.smarthome.automation.Condition@5ef0d29e, of rule: myRule21-1546347186
10:08:52.076 [main] DEBUG o.e.s.a.c.i.c.CompositeModuleHandlerFactory - Set handler to composite module:org.eclipse.smarthome.automation.Condition@5ef0d29e -> org.eclipse.smarthome.automation.core.internal.composite.CompositeConditionHandler@51850751
10:08:52.076 [main] DEBUG o.e.s.a.c.i.c.CompositeModuleHandlerFactory - Set handler to composite module:org.eclipse.smarthome.automation.core.internal.RuntimeCondition@6ca0256d -> org.eclipse.smarthome.automation.core.internal.composite.CompositeConditionHandler@3ce3db41
10:08:52.077 [main] DEBUG o.e.s.c.c.r.AbstractManagedProvider - Added new element myRule21-1546347186 to ManagedRuleProvider.
10:08:52.078 [main] INFO  o.e.s.a.core.internal.RuleEngine - The rule rId = myRule21-1546347186 is already enabled
10:08:52.086 [ESH-safeCall-5] INFO  o.e.s.a.m.c.internal.RuntimeRuleTest - Event: smarthome/items/myMotionItem2/state
10:08:52.086 [pool-3-thread-1] DEBUG o.e.s.a.m.c.h.ItemStateConditionHandler - ItemStateCondition 'ItemStateCondition3'checking if myPresenceItem2 (State=ON) = ON
10:08:52.086 [pool-3-thread-1] DEBUG o.e.s.a.m.c.h.ItemStateConditionHandler - ConditionSatisfied --> true
10:08:52.086 [pool-3-thread-1] DEBUG o.e.s.a.m.c.h.ItemPostCommandActionHandler - Executing ItemPostCommandAction on Item myLampItem2 with Command ON
10:08:52.086 [pool-3-thread-1] DEBUG o.e.s.a.core.internal.RuleEngine - The rule 'myRule21-1546347186' is executed.
10:08:52.087 [ESH-safeCall-5] INFO  o.e.s.a.m.c.internal.RuntimeRuleTest - Event: smarthome/items/myLampItem2/state
10:08:52.187 [main] INFO  o.e.s.a.m.c.internal.RuntimeRuleTest - myLampItem2 State: ON
10:08:52.189 [main] DEBUG o.e.s.c.c.registry.AbstractRegistry - Provider '$Proxy8' has been removed.
10:08:52.190 [main] DEBUG o.e.s.c.c.registry.AbstractRegistry - Provider 'ManagedRuleProvider' has been removed.
10:08:52.190 [main] DEBUG o.e.s.c.c.registry.AbstractRegistry - Provider 'ManagedItemProvider' has been removed.
Tests run: 5, Failures: 1, Errors: 0, Skipped: 1, Time elapsed: 3.766 sec <<< FAILURE! - in org.eclipse.smarthome.automation.module.core.internal.RuntimeRuleTest
assert that compareCondition works(org.eclipse.smarthome.automation.module.core.internal.RuntimeRuleTest)  Time elapsed: 0.153 sec
check if moduleTypes are registered(org.eclipse.smarthome.automation.module.core.internal.RuntimeRuleTest)  Time elapsed: 3.063 sec  <<< FAILURE!
java.lang.AssertionError: 
Expected: is not null
     but: was null
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:865)
	at org.junit.Assert.assertThat(Assert.java:832)
	at org.junit.Assert$assertThat.callStatic(Unknown Source)
	at org.eclipse.smarthome.automation.module.core.internal.RuntimeRuleTest$_check_if_moduleTypes_are_registered_closure14.doCall(RuntimeRuleTest.groovy:199)
	at org.eclipse.smarthome.automation.module.core.internal.RuntimeRuleTest$_check_if_moduleTypes_are_registered_closure14.doCall(RuntimeRuleTest.groovy)
	at sun.reflect.GeneratedMethodAccessor5.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:90)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:324)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:278)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1016)
	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:39)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:108)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:112)
	at org.eclipse.smarthome.test.OSGiTest.waitForAssert(OSGiTest.groovy:181)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:207)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:56)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:49)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:133)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:149)
	at org.eclipse.smarthome.automation.module.core.internal.RuntimeRuleTest.check if moduleTypes are registered(RuntimeRuleTest.groovy:192)

assert that rule is triggered by composite trigger(org.eclipse.smarthome.automation.module.core.internal.RuntimeRuleTest)  Time elapsed: 0.135 sec
assert that item state is updated by simple rule(org.eclipse.smarthome.automation.module.core.internal.RuntimeRuleTest)  Time elapsed: 0.121 sec
testPredefinedRule(org.eclipse.smarthome.automation.module.core.internal.RuntimeRuleTest) skipped

Results :

Failed tests: 
  RuntimeRuleTest.check if moduleTypes are registered:192->OSGiTest.waitForAssert:181 
Expected: is not null
     but: was null

Tests run: 5, Failures: 1, Errors: 0, Skipped: 1

10:08:52.215 [Framework stop] DEBUG o.e.s.c.x.o.XmlDocumentBundleTracker - Removing the XML related objects from module 'org.eclipse.osgi'...
10:08:52.219 [Framework stop] DEBUG o.e.s.c.x.o.XmlDocumentBundleTracker - Removing the XML related objects from module 'org.eclipse.smarthome.automation.module.core.test'...
10:08:52.220 [Framework stop] DEBUG o.e.s.c.x.o.XmlDocumentBundleTracker - Removing the XML related objects from module 'org.eclipse.smarthome.automation.module.core'...
10:08:52.221 [Framework stop] DEBUG o.e.s.c.x.o.XmlDocumentBundleTracker - Removing the XML related objects from module 'org.eclipse.smarthome.config.xml'...
10:08:52.221 [Framework stop] DEBUG o.e.s.c.x.o.XmlDocumentBundleTracker - Removing the XML related objects from module 'org.eclipse.core.runtime'...
10:08:52.221 [Framework stop] DEBUG o.e.s.c.x.o.XmlDocumentBundleTracker - Removing the XML related objects from module 'org.eclipse.smarthome.config.core'...
10:08:52.221 [Framework stop] DEBUG o.e.s.c.x.o.XmlDocumentBundleTracker - Removing the XML related objects from module 'org.eclipse.smarthome.core.autoupdate'...
10:08:52.221 [Framework stop] DEBUG o.e.s.c.x.o.XmlDocumentBundleTracker - Removing the XML related objects from module 'org.eclipse.smarthome.automation.providers'...
10:08:52.221 [Framework stop] DEBUG o.e.s.c.x.o.XmlDocumentBundleTracker - Removing the XML related objects from module 'org.eclipse.smarthome.automation.parser.gson'...
10:08:52.221 [Framework stop] DEBUG o.e.s.c.x.o.XmlDocumentBundleTracker - Removing the XML related objects from module 'org.eclipse.equinox.util'...
10:08:52.221 [Framework stop] DEBUG o.e.s.c.x.o.XmlDocumentBundleTracker - Removing the XML related objects from module 'org.eclipse.tycho.surefire.osgibooter'...
10:08:52.221 [Framework stop] DEBUG o.e.s.c.x.o.XmlDocumentBundleTracker - Removing the XML related objects from module 'org.eclipse.smarthome.automation.core'...
10:08:52.221 [Framework stop] DEBUG o.e.s.c.x.o.XmlDocumentBundleTracker - Removing the XML related objects from module 'org.eclipse.equinox.registry'...
10:08:52.221 [Framework stop] DEBUG o.e.s.c.x.o.XmlDocumentBundleTracker - Removing the XML related objects from module 'org.eclipse.equinox.preferences'...
10:08:52.221 [Framework stop] DEBUG o.e.s.c.x.o.XmlDocumentBundleTracker - Removing the XML related objects from module 'org.eclipse.equinox.event'...
10:08:52.221 [Framework stop] DEBUG o.e.s.c.x.o.XmlDocumentBundleTracker - Removing the XML related objects from module 'org.eclipse.equinox.app'...
10:08:52.221 [Framework stop] DEBUG o.e.s.c.x.o.XmlDocumentBundleTracker - Removing the XML related objects from module 'org.eclipse.equinox.ds'...
10:08:52.221 [Framework stop] DEBUG o.e.s.c.x.o.XmlDocumentBundleTracker - Removing the XML related objects from module 'org.codehaus.groovy'...
10:08:52.221 [Framework stop] DEBUG o.e.s.c.x.o.XmlDocumentBundleTracker - Removing the XML related objects from module 'org.eclipse.smarthome.automation.api'...
10:08:52.221 [Framework stop] DEBUG o.e.s.c.x.o.XmlDocumentBundleTracker - Removing the XML related objects from module 'org.eclipse.core.jobs'...
10:08:52.221 [Framework stop] DEBUG o.e.s.c.x.o.XmlDocumentBundleTracker - Removing the XML related objects from module 'org.hamcrest.core'...
10:08:52.221 [Framework stop] DEBUG o.e.s.c.x.o.XmlDocumentBundleTracker - Removing the XML related objects from module 'org.eclipse.core.contenttype'...
10:08:52.221 [Framework stop] DEBUG o.e.s.c.x.o.XmlDocumentBundleTracker - Removing the XML related objects from module 'org.eclipse.equinox.common'...
10:08:52.222 [Framework stop] DEBUG o.e.s.c.x.o.XmlDocumentBundleTracker - Removing the XML related objects from module 'org.eclipse.smarthome.core'...
10:08:52.354 [Framework stop] DEBUG o.e.s.c.a.i.AutoUpdateActivator - AutoUpdate binding has been stopped.
10:08:52.356 [Framework stop] DEBUG o.e.s.core.internal.CoreActivator - Core bundle has been stopped.
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Eclipse SmartHome .................................. SUCCESS [ 29.078 s]
[INFO] Eclipse SmartHome Bundles .......................... SUCCESS [  0.027 s]
[INFO] Eclipse SmartHome Test Components .................. SUCCESS [  0.014 s]
[INFO] Eclipse SmartHome Core Components .................. SUCCESS [  0.016 s]
[INFO] Eclipse SmartHome Core ............................. SUCCESS [  2.662 s]
[INFO] Eclipse SmartHome Configurations ................... SUCCESS [  0.011 s]
[INFO] Eclipse SmartHome Configuration Core ............... SUCCESS [  0.530 s]
[INFO] Eclipse SmartHome Configuration XML ................ SUCCESS [  0.475 s]
[INFO] Eclipse SmartHome AutoUpdate Binding ............... SUCCESS [  0.126 s]
[INFO] Eclipse SmartHome Test ............................. SUCCESS [  1.165 s]
[INFO] Eclipse SmartHome Configuration Dispatcher ......... SUCCESS [  0.166 s]
[INFO] Eclipse SmartHome Configuration Core Tests ......... SUCCESS [  3.432 s]
[INFO] Eclipse SmartHome I/O Components ................... SUCCESS [  0.011 s]
[INFO] Eclipse SmartHome Console .......................... SUCCESS [  0.146 s]
[INFO] Eclipse SmartHome Thing ............................ SUCCESS [  0.805 s]
[INFO] Eclipse SmartHome Configuration Discovery .......... SUCCESS [  0.310 s]
[INFO] Eclipse SmartHome Configuration Discovery Tests .... SUCCESS [  8.468 s]
[INFO] Eclipse SmartHome Config XML Tests ................. SUCCESS [  3.363 s]
[INFO] Eclipse SmartHome Sample Extension Service ......... SUCCESS [  0.145 s]
[INFO] Eclipse SmartHome mDNS Service ..................... SUCCESS [  0.145 s]
[INFO] Eclipse SmartHome REST Interface ................... SUCCESS [  0.214 s]
[INFO] Eclipse SmartHome Core ID .......................... SUCCESS [  0.148 s]
[INFO] Eclipse SmartHome Thing Tests ...................... SUCCESS [  2.363 s]
[INFO] Eclipse SmartHome Core Tests ....................... SUCCESS [  4.741 s]
[INFO] Eclipse SmartHome Thing Tests ...................... SUCCESS [  6.535 s]
[INFO] Eclipse SmartHome Core Persistence ................. SUCCESS [  0.096 s]
[INFO] Eclipse SmartHome Core Scheduler Service ........... SUCCESS [  0.167 s]
[INFO] Eclipse SmartHome Network I/O ...................... SUCCESS [  0.149 s]
[INFO] Eclipse SmartHome Models ........................... SUCCESS [  0.012 s]
[INFO] Eclipse SmartHome Model Core ....................... SUCCESS [  0.270 s]
[INFO] Eclipse SmartHome Model Lazy Generation ............ SUCCESS [  0.218 s]
[INFO] Eclipse SmartHome Model Items ...................... SUCCESS [ 12.800 s]
[INFO] Eclipse SmartHome Model Persistence ................ SUCCESS [  5.411 s]
[INFO] Eclipse SmartHome Model Script ..................... SUCCESS [ 12.172 s]
[INFO] Eclipse SmartHome Core Transformation Service ...... SUCCESS [  0.178 s]
[INFO] Eclipse SmartHome Core Binding XML ................. SUCCESS [  0.156 s]
[INFO] Eclipse SmartHome Core Binding XML Tests ........... SUCCESS [  3.298 s]
[INFO] Eclipse SmartHome Core Thing XML ................... SUCCESS [  0.217 s]
[INFO] Eclipse SmartHome Core Thing XML Tests ............. SUCCESS [  4.288 s]
[INFO] Eclipse SmartHome Automation ....................... SUCCESS [  0.009 s]
[INFO] Eclipse SmartHome Automation API ................... SUCCESS [  0.211 s]
[INFO] Eclipse SmartHome Automation Core .................. SUCCESS [  0.249 s]
[INFO] Eclipse SmartHome Automation Commands .............. SUCCESS [  0.213 s]
[INFO] Eclipse SmartHome Automation Parser GSON ........... SUCCESS [  0.150 s]
[INFO] Eclipse SmartHome Automation Providers ............. SUCCESS [  0.219 s]
[INFO] Eclipse SmartHome Automation Core Tests ............ SUCCESS [  1.950 s]
[INFO] Eclipse SmartHome Automation Sample Handler ........ SUCCESS [  0.126 s]
[INFO] Eclipse SmartHome Automation Core Modules .......... SUCCESS [  0.184 s]
[INFO] Eclipse SmartHome Automation Core Module Tests ..... FAILURE [  6.091 s]
[INFO] Eclipse SmartHome Automation Timer Modules ......... SKIPPED
[INFO] Eclipse SmartHome Automation Module Timer Tests .... SKIPPED
[INFO] Eclipse SmartHome Automation Script Modules ........ SKIPPED
[INFO] Eclipse SmartHome Automation Script Default Scope .. SKIPPED
[INFO] Eclipse SmartHome Automation Script Module Tests ... SKIPPED
[INFO] Eclipse SmartHome Automation Event Tests ........... SKIPPED
[INFO] Eclipse SmartHome Automation Integration Tests ..... SKIPPED
[INFO] Eclipse SmartHome Automation Java API Demo ......... SKIPPED
[INFO] Eclipse SmartHome Core REST Interface .............. SKIPPED
[INFO] Eclipse SmartHome Automation REST API .............. SKIPPED
[INFO] Eclipse SmartHome Designer Bundles ................. SKIPPED
[INFO] Eclipse SmartHome Designer Core Plugin ............. SKIPPED
[INFO] Eclipse SmartHome Designer UI Plugin ............... SKIPPED
[INFO] Eclipse SmartHome Console for OSGi framework Eclipse Equinox SKIPPED
[INFO] Eclipse SmartHome Console for OSGi Console RFC 147 . SKIPPED
[INFO] Eclipse SmartHome Console for OSGi runtime Karaf ... SKIPPED
[INFO] Eclipse SmartHome Monitor .......................... SKIPPED
[INFO] Eclipse SmartHome Multimedia I/O ................... SKIPPED
[INFO] Eclipse SmartHome Network I/O Tests ................ SKIPPED
[INFO] Eclipse SmartHome IO REST Tests .................... SKIPPED
[INFO] Eclipse SmartHome IO REST Core Tests ............... SKIPPED
[INFO] Eclipse SmartHome SSE Interface .................... SKIPPED
[INFO] Eclipse SmartHome IO SSE Tests ..................... SKIPPED
[INFO] Eclipse SmartHome Model Sitemap .................... SKIPPED
[INFO] Eclipse SmartHome UIs .............................. SKIPPED
[INFO] Eclipse SmartHome UI ............................... SKIPPED
[INFO] Eclipse SmartHome Sitemap REST Interface ........... SKIPPED
[INFO] Eclipse SmartHome MQTT Transport ................... SKIPPED
[INFO] Eclipse SmartHome UPnP Transport ................... SKIPPED
[INFO] Eclipse SmartHome IO UPnP Tests .................... SKIPPED
[INFO] Eclipse SmartHome Model Items Runtime .............. SKIPPED
[INFO] Eclipse SmartHome Model Item UI .................... SKIPPED
[INFO] Eclipse SmartHome Items Model Tests ................ SKIPPED
[INFO] Eclipse SmartHome Model Persistence Runtime ........ SKIPPED
[INFO] Eclipse SmartHome Model Persistence UI ............. SKIPPED
[INFO] Eclipse SmartHome Model Persistence Tests .......... SKIPPED
[INFO] Eclipse SmartHome Model Rules ...................... SKIPPED
[INFO] Eclipse SmartHome Model Rules Runtime .............. SKIPPED
[INFO] Eclipse SmartHome Model Script UI .................. SKIPPED
[INFO] Eclipse SmartHome Model Rules UI ................... SKIPPED
[INFO] Eclipse SmartHome Model Rules Tests ................ SKIPPED
[INFO] Eclipse SmartHome Model Script Runtime ............. SKIPPED
[INFO] Eclipse SmartHome Model Script Tests ............... SKIPPED
[INFO] Eclipse SmartHome Model Sitemap Runtime ............ SKIPPED
[INFO] Eclipse SmartHome Model Sitemap UI Plugin .......... SKIPPED
[INFO] Eclipse SmartHome Thing Model ...................... SKIPPED
[INFO] Eclipse SmartHome Thing Model Runtime .............. SKIPPED
[INFO] Eclipse SmartHome Thing Model UI ................... SKIPPED
[INFO] Eclipse SmartHome Thing Model Tests ................ SKIPPED
[INFO] Eclipse SmartHome UI Tests ......................... SKIPPED
[INFO] Eclipse SmartHome UI Icon Support .................. SKIPPED
[INFO] Eclipse SmartHome UI Icon Tests .................... SKIPPED
[INFO] Eclipse SmartHome Storages ......................... SKIPPED
[INFO] Eclipse SmartHome MapDB Storage .................... SKIPPED
[INFO] Eclipse SmartHome Storage MapDB Tests .............. SKIPPED
[INFO] Eclipse SmartHome Extensions ....................... SKIPPED
[INFO] Eclipse SmartHome Bindings ......................... SKIPPED
[INFO] Eclipse SmartHome hue Binding ...................... SKIPPED
[INFO] Eclipse SmartHome hue Binding Tests ................ SKIPPED
[INFO] Eclipse SmartHome FSInternetRadio Binding .......... SKIPPED
[INFO] Eclipse SmartHome LIFX Binding ..................... SKIPPED
[INFO] Eclipse SmartHome NTP Binding ...................... SKIPPED
[INFO] Eclipse SmartHome Sonos Binding .................... SKIPPED
[INFO] Eclipse SmartHome WeMo Binding ..................... SKIPPED
[INFO] Eclipse SmartHome YahooWeather Binding ............. SKIPPED
[INFO] Eclipse SmartHome Transformation Services .......... SKIPPED
[INFO] Eclipse SmartHome Exec Transform Service ........... SKIPPED
[INFO] Eclipse SmartHome Javascript Transform Service ..... SKIPPED
[INFO] Eclipse SmartHome Map Transform Service ............ SKIPPED
[INFO] Eclipse SmartHome Map Transformation Service Tests . SKIPPED
[INFO] Eclipse SmartHome Scale Transform Service .......... SKIPPED
[INFO] Eclipse SmartHome Scale Transformation Service Tests SKIPPED
[INFO] Eclipse SmartHome RegEx Transform Service .......... SKIPPED
[INFO] Eclipse SmartHome RegEx Transformation Service Tests SKIPPED
[INFO] Eclipse SmartHome XPath Transform Service .......... SKIPPED
[INFO] Eclipse SmartHome XPath Transformation Service Tests SKIPPED
[INFO] Eclipse SmartHome JSonPath Transform Service ....... SKIPPED
[INFO] Eclipse SmartHome JSonPath Transformation Service Tests SKIPPED
[INFO] Eclipse SmartHome Xslt Transform Service ........... SKIPPED
[INFO] Eclipse SmartHome Xslt Transformation Service Tests  SKIPPED
[INFO] Eclipse SmartHome UI Extensions .................... SKIPPED
[INFO] Eclipse SmartHome Icon Sets ........................ SKIPPED
[INFO] Eclipse SmartHome Classic Icon Set ................. SKIPPED
[INFO] Eclipse SmartHome Basic UI ......................... SKIPPED
[INFO] Eclipse SmartHome Classic UI ....................... SKIPPED
[INFO] Eclipse SmartHome Paper UI ......................... SKIPPED
[INFO] Eclipse SmartHome Features ......................... SKIPPED
[INFO] Eclipse SmartHome Runtime Core Feature ............. SKIPPED
[INFO] Eclipse SmartHome Runtime Model Feature ............ SKIPPED
[INFO] Eclipse SmartHome Runtime UI Feature ............... SKIPPED
[INFO] Eclipse SmartHome Runtime REST Feature ............. SKIPPED
[INFO] Eclipse SmartHome Runtime Automation Feature ....... SKIPPED
[INFO] Eclipse SmartHome Runtime Equinox Console Feature .. SKIPPED
[INFO] Eclipse SmartHome Runtime RfC147 Console Feature ... SKIPPED
[INFO] Eclipse SmartHome Runtime Binding Feature .......... SKIPPED
[INFO] Eclipse SmartHome Designer Dependencies ............ SKIPPED
[INFO] Eclipse SmartHome Designer Feature ................. SKIPPED
[INFO] Eclipse SmartHome Test Dependencies ................ SKIPPED
[INFO] Eclipse SmartHome Karaf Feature .................... SKIPPED
[INFO] Eclipse SmartHome Products ......................... SKIPPED
[INFO] Eclipse SmartHome Designer RCP Application ......... SKIPPED
[INFO] Eclipse SmartHome p2 Repository .................... SKIPPED
[INFO] Eclipse SmartHome Tools ............................ SKIPPED
[INFO] Eclipse SmartHome Archetypes ....................... SKIPPED
[INFO] Eclipse Smarthome Binding Archetype ................ SKIPPED
[INFO] Eclipse Smarthome Binding Archetype Testsuite ...... SKIPPED
[INFO] Eclipse SmartHome Target Platform .................. SKIPPED
[INFO] Eclipse SmartHome Distribution ..................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 02:58 min
[INFO] Finished at: 2016-01-29T10:08:52+01:00
[INFO] Final Memory: 251M/958M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-surefire-plugin:0.23.1:test (default-test) on project org.eclipse.smarthome.automation.module.core.test: There are test failures.
[ERROR] 
[ERROR] Please refer to /home/maggu2810/bin/pkgs/eclipse/smarthome-master/git/smarthome/bundles/automation/org.eclipse.smarthome.automation.module.core.test/target/surefire-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :org.eclipse.smarthome.automation.module.core.test
```

--- a/bundles/automation/org.eclipse.smarthome.automation.module.core/ESH-INF/automation/moduletypes/GenericCondition.json
+++ b/bundles/automation/org.eclipse.smarthome.automation.module.core/ESH-INF/automation/moduletypes/GenericCondition.json
@@ -24,12 +24,7 @@
           "type":"TEXT",
           "description":"the compare operator, allowed are <, >, =, matches",
           "required":true,
-          "defaultValue":"=",
-          "options":{
-          "=":"=",
-          "<":"<",
-          ">":">"
-          }
+          "defaultValue":"="
         }
       ],
       "inputs": [

